demo-app/src/pages/platform-devices/index.jsx

```diff
@@ -0,0 +1,223 @@
+    };
+  }),
+};
+
+const devicesWithLocation = {
+  ...devices,
+  features: devices.features.filter(device => {
+    if (device.geometry.coordinates[0] !== null) {
+      return device;
+    }
+  }),
+};
+
+const devicesWithoutLocation = {
+  ...devices,
+  features: devices.features.filter(device => {
+    if (device.geometry.coordinates[0] === null) {
+      return device;
+    }
+  }),
+};
+
+const properties = {
+  'properties.serial_number': {
+    label: 'Serial number',
+    search: true,
+  },
+  'properties.device_type': {
+    label: 'Device type',
+    search: true,
+  },
+  'properties.name': {
+    label: 'Name',
+    search: true,
+  },
+  'properties.ip_address': {
+    label: 'IP address',
+    search: true,
+  },
+  'properties.mac': {
+    label: 'MAC address',
+    search: true,
+  },
+  'properties.part_number': {
+    label: 'Part number',
+    search: true,
+  },
+  'properties.geo.city': {
+    label: 'City',
+    search: true,
+  },
+  'properties.geo.state': {
+    label: 'State',
+    search: true,
+  },
+  'properties.geo.country': {
+    label: 'Country',
+    search: true,
+  },
+  'properties.location': {
+    label: 'Location',
+    search: true,
+  },
+};
+
+const columns = [
+  {
+    property: 'properties.serial_number',
+    header: 'Serial number',
+    primary: true,
+  },
+  {
+    property: 'properties.device_type',
+    header: 'Device type',
+  },
+  { property: 'properties.mac', header: 'MAC address' },
+  { property: 'properties.ip_address', header: 'IP address' },
+  { property: 'properties.name', header: 'Location name' },
+  { property: 'properties.geo.city', header: 'City' },
+  { property: 'properties.geo.state', header: 'State' },
+  { property: 'properties.geo.country', header: 'Country' },
+];
+
+const noLocationMessage = `${new Intl.NumberFormat(navigator.language).format(
+  devicesWithoutLocation.features.length,
+)} devices do not have location specified. ${new Intl.NumberFormat(
+  navigator.language,
+).format(devicesWithLocation.features.length)} of ${new Intl.NumberFormat(
+  navigator.language,
+).format(
+  devicesWithLocation.features.length + devicesWithoutLocation.features.length,
+)} devices displayed.`;
+
+const PlatformDevices = () => {
+  const containerRef = React.useRef();
+  // const [loading, setLoading] = React.useState(true);
+  const [view, setView] = React.useState('map');
+
+  // React.useEffect(() => {
+  //   setTimeout(() => {
+  //     setLoading(false);
+  //   }, 1000);
+  // }, []);
+
+  return (
+    <Page fill>
+      <PageContent flex>
+        <PageHeader
+          title="Acme, Inc. Devices"
+          parent={<ReverseAnchor as={Link} label="Home" to="/" />}
+        />
+        <ContentContainer flex={false}>
+          <Data data={devicesWithLocation.features} properties={properties}>
+            <Box gap="small">
+              <Toolbar direction="column" gap="xsmall">
+                <ToolbarRegion>
+                  <ToolbarRegion gap="small">
```

Comment:
Nested toolbar regions makes me pause

Reply:
Say more.

Would the following be preferable?
```
<ToolbarRegion>
  <Box direction="row" gap="small">
```

Reply:
I'm not opposed to having nested regions, but was trying to map them to a semantic purpose and my initial feeling was that they would be peers of each other, children of Toolbar.

I do think there's utility to not specify Box/row again

Reply:
Semantically, the outer toolbarregions represent 'view controls' and 'action controls'. Within these regions (at least within view controls) there are sub regions of 'find controls', 'saved views', and 'customization controls.' Because of the affinity the controls within these sub regions share, the gaps between their inner controls should be smaller than the gaps between the sub regions themselves.

Reply:
I'm fine with this pattern for now.